cdn/common/memory-alert.py

```
@@ -0,0 +1,213 @@
+            return int(f.read().strip()) // (1024 * 1024)
+    except (OSError, ValueError):
+        return 0
+
+
+def send_webhook(url, message):
+    """Post a message to a Discord/Slack webhook."""
+    payload = json.dumps({"content": message}).encode()
+    req = urllib.request.Request(
+        url, data=payload,
+        headers={"Content-Type": "application/json"},
+    )
```

Ruff (0.15.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)